Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Last-Update: 2012-06-28
Forwarded: not-needed
Description: Disable supression of fortify hardening flag.
 Wirh --enable-debug upstream build system removes '-02' argument
 from CFLAGS and therefore prevent fortification.
 This patch disable dropping build flags from CFLAGS.
 .
 Note: although according to blhc (see http://ruderich.org/simon/blhc/)
 fortify flag is present there are still some real false-positives
 from lintian i.e. W:hardening-no-fortify-functions.

--- a/configure.ac
+++ b/configure.ac
@@ -621,10 +621,6 @@
 AC_ARG_ENABLE( debug,
   [AS_HELP_STRING([--enable-debug],[compile with debugging flags set])],
   [
-     # remove any optimization flags...
-     CFLAGS=`echo ${CFLAGS} | sed -e 's,-O.,,g'`
-     # ...except for those the user wants.
-     CFLAGS="${CFLAGS} -g ${USER_OPTIMIZATION}"
      LDFLAGS="${LDFLAGS} -g"
      AC_DEFINE(DEBUG_MEMORY,1,[Enable debug memory])
      AC_DEFINE(DEBUG_MEMORY,0,[Enable debug memory])
